### PR TITLE
add node affinity to linkerd-viz

### LINKED ISSUE #10680: linkerd-viz and linkerd-jaeger charts do not support nodeAffinity
### What is the issue?

In contrast to [linkerd-control-plane](https://github.com/linkerd/linkerd2/blob/main/charts/linkerd-control-plane/templates/proxy-injector.yaml#L58) helm chart `linkerd-viz` and `linkerd-jaeger` helm charts do not support nodeAffinity at the moment. Can you please add this functionality there unless there is any specific reason not to do it?

### How can it be reproduced?

Deploy latest version of linkerd-viz and linkerd-jaeger

### Logs, error output, etc

no errors

### output of `linkerd check -o short`

not required

### Environment

kubernetes 1.25.5
linkerd 2.12.4

### Possible solution

update helm charts to add nodeAffinity

### Additional context

_No response_

### Would you like to work on fixing this bug?

None

### PR DESCRIPTION
**Pull request by reyes256**
Add to Helm linkerd-viz and tracing deployment resources nodeAffinity labels

NodeAffinity was not enabled for Viz Deployments resources

Add to helm values and each of Viz Deployment the labels to add the affinity labels from the shared partials and
allow to be more granular on the manifest setup

cd viz/charts/linkerd-viz/
helm template .

Fixes #10680

Signed-off-by: Fernando Reyes hapoweiser@gmail.com

### CODE REVIEW

File: viz/charts/linkerd-viz/templates/metrics-api.yaml [hunk 1]
```diff
@@ -71,9 +71,10 @@ spec:
       {{- include "linkerd.tolerations" (dict "Values" .Values.metricsAPI) | nindent 6 }}
       {{- end -}}
       {{- include "linkerd.node-selector" (dict "Values" .Values.metricsAPI) | nindent 6 }}
+      {{- include "linkerd.affinity" (dict "Values" .Values.metricsAPI) | nindent 6 }}
```
Comment (alpeb) [inline viz/charts/linkerd-viz/templates/metrics-api.yaml:74]: Why was this line moved?

**Review by alpeb — COMMENTED** [review]
Thanks @reyes256 :+1: 

The PR description mentions Jaeger, but the changes are about linkerd-viz. Can you amend that?

Viz' `values.yaml` already contains a global `nodeAffinity` entry (commented out). Can you remove it?
For completeness sake, can you add `nodeAffinity` to the tap, tapInjector and dashboard components as well?

**Comment by alpeb** [comment]
Hi @reyes256 , do you still want to take this one across the finish line? Is there anything I can help you with?

**Comment by reyes256** [comment]
@alpeb yes i would like to continue, i'll be working on it

File: viz/charts/linkerd-viz/templates/metrics-api.yaml (hunk 1 above)
Comment (reyes256) [reply viz/charts/linkerd-viz/templates/metrics-api.yaml:74]: Somehow the changes didn't work accordingly, and adding a line break at the end solved the problem

**Comment by reyes256** [comment]
@alpeb Seems like 'nodeAffinity' has already been added in both tap and tap-injector components, however i'm not sure how to modify the dashboard component, its not in a separate file and i dont know if its directly on the values.yml

File: viz/charts/linkerd-viz/templates/metrics-api.yaml (hunk 1 above)
Comment (alpeb) [reply viz/charts/linkerd-viz/templates/metrics-api.yaml:74]: I wan't able to reproduce the issue. Can you provide more details? (I don't understand the comment about empty component label).

**Review by alpeb — COMMENTED** [review]
> @alpeb Seems like 'nodeAffinity' has already been added in both tap and tap-injector components

They're calling the `linkerd.affinity` partial, but there's no `tap.nodeAffinity` nor `tapInjector.nodeAffinity` entries in `values.yaml`.

> however i'm not sure how to modify the dashboard component, its not in a separate file and i dont know if its directly on the values.yml

It's the `web.yaml` template, that uses the `dashboard` entries from `values.yaml`

**Comment by alpeb** [comment]
This PR appears to be stale, so closing for now. Please reopen a new one addressing the feedback above if you're still interested on implementing this.
